lib/compiler/src/translator/sections.rs

```diff
@@ -327,8 +303,24 @@ fn read_elems(items: &ElementItems) -> WasmResult<Box<[FunctionIndex]>> {
     let mut elems = Vec::with_capacity(usize::try_from(items_reader.get_count()).unwrap());
     for item in items_reader {
         let elem = match item? {
-            ElementItem::Null(_ty) => FunctionIndex::reserved_value(),
             ElementItem::Func(index) => FunctionIndex::from_u32(index),
+            ElementItem::Expr(expr) => {
+                let mut e = expr.get_operators_reader();
+                let op = e.read()?;
+                let end = e.read()?;
+                e.ensure_end()?;
+                match (op, end) {
+                    (Operator::RefFunc { function_index }, Operator::End) => {
+                        FunctionIndex::from_u32(function_index)
+                    }
+                    (Operator::RefNull { .. }, Operator::End) => FunctionIndex::reserved_value(),
+                    _ => {
+                        return Err(WasmError::Generic(String::from(
+                            "unexpected elem initializer",
+                        )))
+                    }
+                }
```

Comment:
We are gonna need a proper const expression evaluator, this is pretty yucky and, I think, will not work for some of const expressions that are still valid-ish.

Reply:
Hmmmm does it? My understanding of the "Constant expressions" paragraph of the spec is that it must be composed only of `t.const`, `ref.null`, `ref.func` and `global.get` where the global is imported and constant.

So here, I think only these two sequences of operations can occur?